packages/flutter/lib/src/material/toggle_buttons.dart

```diff
@@ -667,64 +668,147 @@ class ToggleButtons extends StatelessWidget {
       final BorderSide borderSide = _getBorderSide(index, theme, toggleButtonsTheme);
       final BorderSide trailingBorderSide = _getTrailingBorderSide(index, theme, toggleButtonsTheme);
 
-      return _ToggleButton(
-        selected: isSelected[index],
-        textStyle: textStyle,
-        constraints: constraints,
-        color: color,
-        selectedColor: selectedColor,
-        disabledColor: disabledColor,
-        fillColor: fillColor,
-        focusColor: focusColor ?? toggleButtonsTheme.focusColor,
-        highlightColor: highlightColor ?? toggleButtonsTheme.highlightColor,
-        hoverColor: hoverColor ?? toggleButtonsTheme.hoverColor,
-        splashColor: splashColor ?? toggleButtonsTheme.splashColor,
-        focusNode: focusNodes != null ? focusNodes![index] : null,
-        onPressed: onPressed != null
-          ? () {onPressed!(index);}
-          : null,
-        mouseCursor: mouseCursor,
+      final Set<MaterialState> states = <MaterialState>{
+          if (isSelected[index] && onPressed != null) MaterialState.selected,
+          if (onPressed == null) MaterialState.disabled,
+      };
+      final Color effectiveFillColor = _ResolveFillColor(fillColor
+        ?? toggleButtonsTheme.fillColor).resolve(states)
+        ?? _DefaultFillColor(theme.colorScheme).resolve(states);
+      final Color currentColor;
+      if (onPressed != null && isSelected[index]) {
+        currentColor = selectedColor
+          ?? toggleButtonsTheme.selectedColor
+          ?? theme.colorScheme.primary;
+      } else if (onPressed != null && !isSelected[index]) {
+        currentColor = color
+          ?? toggleButtonsTheme.color
+          ?? theme.colorScheme.onSurface.withOpacity(0.87);
+      } else {
+        currentColor = disabledColor
+          ?? toggleButtonsTheme.disabledColor
+          ?? theme.colorScheme.onSurface.withOpacity(0.38);
+      }
+      final TextStyle currentTextStyle = textStyle
+        ?? toggleButtonsTheme.textStyle
+        ?? theme.textTheme.bodyText2!;
+      final BoxConstraints? currentConstraints = constraints
+        ?? toggleButtonsTheme.constraints;
+      final Size minimumSize = currentConstraints == null
+        ? const Size.square(kMinInteractiveDimension)
+        : Size(currentConstraints.minWidth, currentConstraints.minHeight);
+      final Size? maximumSize = currentConstraints == null
+        ? null
+        : Size(currentConstraints.maxWidth, currentConstraints.maxHeight);
+      final Size minPaddingSize;
+      switch (tapTargetSize ?? theme.materialTapTargetSize) {
+        case MaterialTapTargetSize.padded:
+          if (direction == Axis.horizontal) {
+            minPaddingSize = const Size(
+              kMinInteractiveDimension,
```

Comment:
```suggestion
              0.0,
```

Reply:
My understanding was input padding height needed to match 48.0 when regardless of the custom [constraints](https://api.flutter.dev/flutter/material/ToggleButtons/constraints.html)  but I was probably mistaken and the accessibility green window should take input padding space.

![image](https://user-images.githubusercontent.com/48603081/164681996-2732beac-e5ec-4e70-a17e-b1d1e9b92767.png)


To be double sure, I just checked the native reference and it does look like input padding is not supposed to be higher than the toggle button.

![2022-04-22 12 41 08](https://user-images.githubusercontent.com/48603081/164682020-3dcfc39d-2fd7-4994-9ea4-30729a68bf7a.jpg)



This is fixed by this suggestion provided here.


![2022-04-22 12 41 03](https://user-images.githubusercontent.com/48603081/164682029-26aa0371-ef3d-418f-ad9f-fa217b177b24.jpg)

Reply:
The input padding is supposed to be higher than the toggle button. Even in the screenshots provided here, you can see the difference. In the middle screenshot, the green rectangle is taller than the "everyone" button, while in the bottom screenshot the green rectangle is fully contained within the "locks" button. Here is a screenshot highlighting the exact bounds of the "everyone" button:

![image](https://user-images.githubusercontent.com/4040946/164713826-4b71a33e-20e8-4451-804b-336ab08c14c9.png)

Note the slight difference in height between the rendering and the actual bounds.

Reply:
Thank for highlighting this, the semantic focus window indeed looks slightly taller but not `48.0` from the looks of it.

If we set `kMinInteractiveDimension` , it would rather look like this. 
![image](https://user-images.githubusercontent.com/48603081/164718368-4c220302-4bc6-492f-b3c2-583be0092047.png)


Than this 

![image](https://user-images.githubusercontent.com/48603081/164718390-ff8f403f-39d6-44b6-bc00-655ab34dbfa4.png)


 Could it be that the focus window is wrapping like an outer border in the native app and wrapping like an inner border in Flutter?.

You can see my updated screenshot, green border looks like an inner border? and the native focus window looks like an outer border. 

Since it is not a standard height like `48.0` so what height it should be? or vertical padding like `4.0` something? I don't see such a design in other widgets. 


`Chip` has such padding and it is `kMinInteractiveDimension`.
https://github.com/flutter/flutter/blob/fde7622f7e0ab915ce19bd49a8ead5707141df6a/packages/flutter/lib/src/material/chip.dart#L1241-L1252



![2022-04-22 15 53 12](https://user-images.githubusercontent.com/48603081/164718226-61f3bab1-0102-422f-ad30-386fc49aa77a.jpg)
![2022-04-22 15 53 16](https://user-images.githubusercontent.com/48603081/164718231-857a7558-57cf-45b1-b348-cfac1bc7aa2c.jpg)

Reply:
I tested this against your project's semantics tests yesterday @math1man, and I believe this produces the desired behavior. I will send you the report. I know we originally thought we solved this in https://github.com/flutter/flutter/issues/93249, so I want o make sure we get it right this time.

Reply:
@TahaTesser l#708 is the width, isn't it? It looks like l#709 should be `kMinInteractiveDimension` for the height.

Reply:
My apologies just fixed this.

Reply:
No worries at all! :)